### DISCUSSION TITLE
Different quote_dropped_files for (n)vim only on Windows

### DISCUSSION THREAD

**Discussion by Grueslayer** [wezterm/wezterm#6181] [General]
Hi,

on windows we need double quoted filenames as usual but (neo)vim needs some other format.

Can we make that option "per tab" and configurable via callback:

if we've a callback where 
- I can get the original `pane_title` (not modified by `format-tab-title`), i can decide (neo)vim is running (foreground_process_name can not be used because (neo)vim has spawned node/python or others)
- I would return `SpacesOnly` (looks weird, but `c:\dir\ with\ space\file.txt`  is valid)

Maybe we can define a custom lua function which does the quotation itself (but still needs access to `pane_title`)

**Comment by wez** [comment]
you can use https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html from inside the https://wezfurlong.org/wezterm/config/lua/window-events/update-status.html event to make decisions about the running program and dynamically adjust the configuration

**Reply by Grueslayer** [reply]
```lua
wezterm.on('update-status', function(window, pane)
  local overrides = window:get_config_overrides() or {}
  overrides.enable_scroll_bar = not pane:is_alt_screen_active()

  if wezterm.target_triple ==  "x86_64-pc-windows-msvc"  then
    if pane:get_title():match(" - nvim$") then
      overrides.quote_dropped_files = "SpacesOnly"
    else
      overrides.quote_dropped_files = nil
    end
  end

  window:set_config_overrides(overrides)
end)

```

and set title in (neo)vim

```lua
vim.o.titlestring = '%t%( %M%)%( %{expand("%:~:.:h")}%)%( %a%) - nvim'
```
